_overviews/scala3-book/domain-modeling-fp.md

```diff
@@ -0,0 +1,408 @@
+---
+title: FP Modeling
+type: section
+description: This chapter provides an introduction to FP domain modeling with Scala 3.
+num: 22
+previous-page: domain-modeling-oop
+next-page: methods-intro
+---
+
+
+This chapter provides an introduction to functional programming (FP) style domain modeling in Scala 3. When modeling the world around us with FP, you typically use these Scala constructs:
+
+- Enumerations
+- Case classes
+- Traits
+
+
+## Introduction
+
+In FP, the *data* and the *operators on that data* are two separate things; you aren’t forced to encapsulate them together like you do with OOP.
+
+The concept is like numerical algebra. When you think about whole numbers whose values are greater than or equal to zero, you have a *set* of possible values that looks like this:
```

Comment:
Aren't whole numbers by definition greater or equal than 0 ?

Reply:
no whole numbers just means not floating point